datahub-web-react/src/app/entity/shared/tabs/Documentation/components/DescriptionEditor.tsx

```diff
@@ -65,19 +79,45 @@ export const DescriptionEditor = ({ onComplete }: { onComplete?: () => void }) =
         refetch?.();
     };
 
+    // Function to handle all changes in Editor
+    const handleEditorChange = (editedDescription: string) => {
+        setUpdatedDescription(editedDescription);
+        if (editedDescription === description) {
+            setIsDescriptionUpdated(false);
+        } else {
+            setIsDescriptionUpdated(true);
+        }
+    };
+
+    // Updating the localStorage when the user has paused for 5 sec
+    React.useEffect(() => {
+        let delayDebounceFn: ReturnType<typeof setTimeout>;
+        const editedDescriptionsLocal = (localStorageDictionary && JSON.parse(localStorageDictionary)) || {};
+
+        if (isDescriptionUpdated) {
```

Comment:
Why do we need to do this if?

If the user is in editing mode, we should always write to local storage. Once they exit editing mode, we should always clear it.